### ISSUE TITLE
cannot use as with default

### ISSUE THREAD

**Issue by ahuret** [outr/profig#29]
Hi ! I'm new in scala and I'm using profig library to handle configuration in my code.

I can't get defaults mecanism working:

```scala
Profig.load(ProfigLookupPath("/config.yaml", FileType.Yaml, LoadType.Merge))
Profig("plop").as[String]("defaultPlop")
```

on compilation there is an error:

```
[error]  found   : String("defaultPlop")
[error]  required: Int
[error]   Profig("plop").as[String]("defaultPlop")
```

Looks it's should work if I understand well the code in https://github.com/outr/profig/blob/478e07cdab445d0ef95b442eea9320743e418b7b/core/shared/src/main/scala/profig/ProfigPath.scala#L42

Have you an idea why it's not working in my case ?

Thanks a lot

**Comment by vdebergue** [comment]
It seems to be an issue with the Scala compiler that understands `Profig("plop").as[String]("defaultPlop")` as

```scala
def as[T]: T = macro ...
// followed by String##apply
def apply(i: Int): Char = s.charAt(i)
```

You could try with `opt` instead: `Profig("plop").opt[String].getOrElse("defaultPlop")

**Comment by ahuret** [comment]
thanks @vdebergue, but it doesn't works, with error 

``` value opt is not a member of profig.ProfigPath```

**Comment by ahuret** [comment]
@vdebergue it works with your solution and `as[String](default)` when I've updated the library with latest version. Thanks for your help ! Closing it

**Comment by darkfrog26** [comment]
Please feel free to offer any feedback on the library as I'm doing a completely new release this week and am open to any requests for new or changes to functionality.

**Comment by ahuret** [comment]
Nice ! I would not miss it. :-)

**Comment by darkfrog26** [comment]
Profig 3.0 is released and the README is updated

**Comment by ahuret** [comment]
@darkfrog26 tried to upgrade to 3.0.1, I had some (documentation) problems that I resolved.

- I didn't knew about `ExecutionContext` (required when calling `Profig.init()`). I think more precisions would be useful.
- https://github.com/outr/profig#loading-files this part is confusing since I thought `File` was a profig type (few lines before we `import profig._`), I had to check in code directly. I think it should be explicit that this is `java.io.File`
